src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
+
+  dead() {
+    const index = Animal.alive.indexOf(this);
+
+    Animal.alive.splice(index, 1);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide(hidden) {
```

Comment:
The 'hide' method should not take any arguments. It should just set the 'hidden' property of the instance to true.